language/move-prover/spec-lang/src/env.rs

```diff
@@ -1940,3 +1976,58 @@ impl<'env> FunctionEnv<'env> {
         )
     }
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct ConstantData {
+    pub loc: Loc,
+    pub name: Symbol,
+    pub type_: Type,
+    pub idx: ConstantPoolIndex,
+}
+
+#[derive(Debug, Clone)]
+pub struct ConstantEnv<'env> {
+    /// Reference to enclosing module.
+    pub module_env: ModuleEnv<'env>,
+
+    /// Reference to the constant data.
+    data: &'env ConstantData,
+}
+
+impl<'env> ConstantEnv<'env> {
+    pub fn get_name(&self) -> Symbol {
+        self.data.name
+    }
+
+    pub fn get_type(&self) -> Type {
+        self.data.type_.clone()
+    }
+
+    pub fn get_value(&self) -> Value {
+        let const_ = self.module_env.get_constant(self.data.idx);
+        let move_value = self.module_env.get_constant_value(const_);
+        Self::translate_from_move_value(&move_value)
+    }
+
+    fn translate_from_move_value(value: &MoveValue) -> Value {
```

Comment:
I'm not sure if here is the best place to put this function.

Reply:
I would put the ast::Value already in the ConstantDecl, then do the translation in `translate.rs`. There should be already functions inside this module to map MoveValue to ast::Value.

Reply:
I don't actually see such function in `translate.rs`. `translate.rs` deals with values of type `EA::Value` and translates them to `ast::Value`.